app/controllers/stores_controller.rb

```diff
@@ -5,27 +5,36 @@ def new
   end
 
   def create
-    @store = StoreCreator.create_store(current_user, store_params)
+    @store = StoreCreator.create_store(current_user, params[:store][:name])
     if @store.save
       flash_success("Store created successfully.")
-      redirect_to dashboard_path
+      redirect_to current_dashboard_path
     else
       flash_errors(@store)
-      render :new
+      redirect_to new_store_path
```

Comment:
this switched from a render to a redirect during debugging, I don't really care which we stick with